packages/react-form/src/hooks/list/hooks/handlers.ts

```diff
@@ -69,7 +70,11 @@ export function useHandlers<Item extends object>(
               }
             },
             reset() {
-              dispatch(resetAction({target}));
+              if (state.isDynamic) {
+                dispatch(resetListAction());
+              } else {
+                dispatch(resetAction({target}));
```

Comment:
Problem with this: it will reset the whole list for each item :/ Maybe we could find a better way to avoid performances caveats?

Reply:
I dont think we should do this because it means you will not be able to reset individual items if you are using a dynamic list. Also I feel like that would be taking away some functionality that might break flows on people already using DynamicList